### PR TITLE
Added possibility to exclude items in search results

### PR DESCRIPTION
**Pull request by biovenger**
One thing that bothered me a bit is the disconnection between common.js and builder.js. Specifically for displaying an item row. It's a bit awkward as now the excluded state has to be passed onto common.js, that should technically not have to know about it.

### CODE REVIEW

File: static/builder.html [hunk 1]
```diff
@@ -374,6 +374,7 @@
                         <div class="checkboxGroup"><input id="excludeTMR5" type="checkbox"/><label>Exclude TMR of base 5* units</label></div>
                         <div class="checkboxGroup" data-server="JP"><input id="excludeSTMR" type="checkbox"/><label>Exclude STMR</label></div>
                         <div class="checkboxGroup"><input id="excludeNotReleasedYet" type="checkbox" checked="true"><label>Exclude not released yet equipments</label></div>
+                        <div><a class="link" onclick="displayEquippableItemList();">Select items to exclude</a></div>
```
Comment (lyrgard) [inline static/builder.html]: Minor : should be displayEquipableItemList, with only one 'p', no ?

File: static/builder.js [hunk 2]
```diff
@@ -1451,8 +1495,16 @@ function displaySearchResultsAsync(items, start, div) {
             if (item.enhancements || itemInventory && itemInventory.enchantments && itemInventory.enchantments[item.id]) {
                 html += " enhanced";
             }
-            html += '" onclick="fixItem(\'' + item.id + '\', ' + currentItemSlot + ', ' + enhancementString + ')">';
-            html += displayItemLine(item);
+            
+            if(searchClickBehavior == ClickBehaviors.EQUIP) {
+                html += '" onclick="fixItem(\'' + item.id + '\', ' + currentItemSlot + ', ' + enhancementString + ')">';
+            } else {
+                html += '" >';
```
Comment (lyrgard) [inline static/builder.js]: Why not take the opportunity to allow to click on the whole line to exclude/include back on item when in the ClickBehaviors.EXCLUDE mode ?

File: static/builder.js [hunk 3]
```diff
@@ -1493,16 +1545,33 @@ function displaySearchResultsAsync(items, start, div) {
 
 function getItemEnhancementLink(item) {
     var html = "";
+    
     if (weaponList.includes(item.type)) {
         html += '<div class="enchantment"><img src="img/dwarf.png" onclick="event.stopPropagation();selectEnchantedItem(\'' + item.id + '\')">';
```
Comment (lyrgard) [inline static/builder.js]: Probably remove the onclick here in EXCLUDE mode, no ?

**Comment by lyrgard** [comment]
regarding common.js, I think you shouldn't push the "exclude state" to it.

The methods to display items in common.js are the part that are common to all pages of FFBE Equip. There are use in the Encyclopedia and in the item inventory. Only changes that should be put there are changes intended to impact all those pages.

You can take a look at the getItemLine function of builder.js. It is the function used to display the build lines. It uses the methodes from common.js for the shared column type (like image and name columns), but adds its own column before them for the various buttons on the left.

File: static/builder.html (hunk 1 above)
Comment (biovenger) [reply static/builder.html]: Good catch!

File: static/builder.js (hunk 2 above)
Comment (biovenger) [reply static/builder.js]: Great suggestion and easy to add, will fix.

File: static/builder.js (hunk 3 above)
Comment (biovenger) [reply static/builder.js]: Maybe remove the functionality entirely when in Exclude mode?

Comment (lyrgard) [reply static/builder.js]: Yes, even better

**Pull request merged**